setup.py

```diff
@@ -14,167 +14,90 @@
-            ext.extra_compile_args = opts
-            ext.extra_link_args = link_opts
-
-        build_ext.build_extensions(self)
 
+        configure_args += self.cmake_defines
+        
+        build_args = []
+
+        if platform.system() == "Darwin":
+            configure_args += []
+        elif platform.system() == "Linux":
+            configure_args += []
+        elif platform.system() == "Windows":
+            configure_args += [
+                "-DENABLE_OPENMP=OFF",
+                "-DENABLE_BLAS=OFF"
+            ]
+        else:
+            raise RuntimeError(f"Unsupported '{platform.system()}' platform")
+
+        if not Path(self.build_temp).exists():
+            os.makedirs(self.build_temp)
+        
+        subprocess.check_call(
+            ["cmake", str(ext.sourcedir)] + configure_args, cwd = self.build_temp
+        )
+        subprocess.check_call(
+            ["cmake", "--build", "."] + build_args, cwd = self.build_temp
+        )
 
-if not os.environ.get("SKIP_COMPILATION", False):
 
-    include_dirs = [
-        get_pybind_include(),
-        "./include",
-        "pennylane_lightning/src/algorithms",
-        "pennylane_lightning/src/simulator",
-        "pennylane_lightning/src/util"
-    ]
+with open("pennylane_lightning/_version.py") as f:
+    version = f.readlines()[-1].split()[-1].strip("\"'")
 
-    library_dirs = [i for i in os.environ.get("LD_LIBRARY_PATH", "").split(":") if i]
-    libraries = []
-    extra_compile_args = []
-    extra_link_args = []
-
-    if os.environ.get("USE_LAPACK", False):
-        extra_compile_args += [" -llapacke -DLAPACKE=1"]
-        libraries += ["lapacke"]
-        extra_link_args += ["-llapacke"]
-
-    if os.environ.get("USE_OPENBLAS", False):
-        extra_compile_args += [" -lopenblas -DLAPACKE=1"]
-        libraries += ["openblas"]
-        extra_link_args += ["-lopenblas"]
-
-    if platform.system() == "Darwin":
-        include_dirs += ["/usr/local/opt/libomp/include"]
-        library_dirs += ["/usr/local/opt/libomp/lib"]
-        libraries += ["omp"]
-
-
-    ext_modules = [
-        Extension(
-            "lightning_qubit_ops",
-            sources=[
-                "pennylane_lightning/src/simulator/StateVector.cpp",
-                "pennylane_lightning/src/algorithms/AdjointDiff.cpp",
-                "pennylane_lightning/src/bindings/Bindings.cpp",
-            ],
-            depends=[
-                "pennylane_lightning/src/algorithms/AdjointDiff.hpp",
-                "pennylane_lightning/src/simulator/StateVector.hpp",
-                "pennylane_lightning/src/util/Util.hpp",
-            ],
-            include_dirs=include_dirs,
-            language="c++",
-            libraries=libraries,
-            library_dirs=library_dirs,
-            extra_compile_args=extra_compile_args,
-            extra_link_args=extra_link_args,
-        ),
-    ]
-else:
-    ext_modules = []
 
 requirements = [
+    "ninja", 
```

Comment:
I assume ninja is a requirement for building on Windows?

Reply:
Ninja works on all platforms (and usually faster than Makefile). I used Ninja as it gives the same build script for all environments. I also could compile the library without Ninja in Linux and MSVC, but it was harder to find out what is the problem (when there is as the output of `CMake` was tons of XML files for msbuild).